episodeFunctions.js

```diff
@@ -0,0 +1,369 @@
+  } else {
+    str += `<article class="episodeArticle">${summary.slice(
+      0,
+      summary.indexOf(" ", maxLength)
+    )}</article>
+			<p class="read" onclick="readMore()">Read more</p>
+			<span style="display: none;">${summary.slice(
+        summary.indexOf(" ", maxLength)
+      )}</span>
+			<p class="read" onclick="readLess()" style="display: none;">Read less</p>`;
+  }
+  // Check if comments already stored in session storage; if they are, render them
+  if (sessionStorage.getItem(`${seriesId}${episodeCode}`) !== null) {
+    let obj = JSON.parse(sessionStorage.getItem(`${seriesId}${episodeCode}`));
+    str += '<p class="episodeComment"><strong>Comments:</strong></p>';
+    for (let i = 0; i < obj.length; i++) {
+      str += `<p class="episodeComment">${obj[i]}</p>`;
+    }
+  }
+  // Add buttons and textbox to add new comments
+  str += `<button class="commentEpisode">Add comment</button>
+			<div class="newComment" style="display: none;"><textarea></textarea><button class="sendComment">Save</button></div>
+			</section>`;
+  return str;
+}
+
+// Add comments
+function addComments(series, color) {
+  let eps = document.querySelectorAll(".episodeSection");
+  for (let i = 0; i < eps.length; i++) {
+    eps[i].style.backgroundColor = color;
+    eps[i].style.display = "none";
+    eps[i].lastElementChild.previousElementSibling.addEventListener(
+      "click",
+      function () {
+        eps[i].lastElementChild.style.display = "flex";
+        // Add event listener to save comment button
+        eps[i].lastElementChild.lastElementChild.addEventListener(
+          "click",
+          function (e) {
+            eps[i].lastElementChild.style.display = "none";
+            let seriesId = series.slice(29, series.indexOf("/episodes"));
+            let txt = e.target.previousElementSibling.value;
+            // If comment is not empty, save it to session storage and render it on the page
+            if (txt.length > 0) {
+              let comment = document.createElement("p");
+              comment.classList = "episodeComment";
+              comment.innerText = txt;
+              let episodeCode = eps[i].id;
+              // If no other comments saved in session storage, create new comment array
+              if (
+                sessionStorage.getItem(`${seriesId}${episodeCode}`) === null
+              ) {
+                let commentArray = [txt];
+                sessionStorage.setItem(
+                  `${seriesId}${episodeCode}`,
+                  JSON.stringify(commentArray)
+                );
+                let commentTitle = document.createElement("p");
+                commentTitle.classList = "episodeComment";
+                commentTitle.innerHTML = "<strong>Comments:</strong>";
+                eps[i].insertBefore(
+                  commentTitle,
+                  eps[i].lastElementChild.previousElementSibling
+                );
+              } else {
```

Comment:
This block of code is starting to get a bit complex (within a function within a function and if statements with lots of indentation)

Can this be split up?